### PR TITLE
Fix Ads enabled for unsupported regions and enabled by default after 0.63.x update

### PR DESCRIPTION
**Pull request by tmancey**
fixes https://github.com/brave/brave-browser/issues/3967
fixes: https://github.com/brave/brave-browser/issues/4131

Several steps were taken to harden the process for retrieving and reacting to whether or not the user's region is supported.

Also hardened were the preference migrations during upgrade, so Ads does not enable on upgrade if it was disabled previously.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Windows
  - [x] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
Confirm Ads UI is only shown for supported regions: US, UK, Canada, France and Germany based upon the operating systems locale.

Additional test plan for upgrade here: https://github.com/brave/brave-browser/issues/4131

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -893,45 +888,50 @@ void RewardsDOMHandler::CheckImported(const base::ListValue *args) {
...
-    }
+  std::string value;
+  args->GetString(1, &value);
 
-    base::ListValue* emptyArgs;
-    GetAdsData(emptyArgs);
+  if (key == "adsEnabled") {
+    ads_service_->SetAdsEnabled(value == "true");
+  } else if (key == "adsPerHour") {
+    ads_service_->SetAdsPerHour(std::stoull(value));
```
Comment (emerick) [inline browser/ui/webui/brave_rewards_ui.cc]: Seems weird that we read these two values as strings and then convert them to other types. Is it not possible to send them as the correct types in the first place?

File: browser/ui/webui/brave_rewards_ui.cc [hunk 2]
```diff
@@ -893,45 +888,50 @@ void RewardsDOMHandler::CheckImported(const base::ListValue *args) {
...
   }
+
+  base::ListValue* emptyArgs;
```
Comment (emerick) [inline browser/ui/webui/brave_rewards_ui.cc]: I know we're not doing anything with this pointer in the function, but maybe we should initialize it to `nullptr` just to play it safe here anyway.

File: vendor/bat-native-ads/README.md [hunk 3]
```diff
@@ -30,6 +30,12 @@ Initialize Ads by calling `Initialize` when Ads are enabled or disabled on the C
 void Initialize()
 ```
 
+`IsSupportedRegion` should be called to determine if Ads are supported for the specified region
+```
+stastic bool IsSupportedRegion(
```
Comment (emerick) [inline vendor/bat-native-ads/README.md]: Typo: stastic

File: browser/ui/webui/brave_rewards_ui.cc (hunk 1 above)
Comment (tmancey) [reply browser/ui/webui/brave_rewards_ui.cc]: @ryanml ^^

File: browser/ui/webui/brave_rewards_ui.cc (hunk 2 above)
Comment (tmancey) [reply browser/ui/webui/brave_rewards_ui.cc]: 👍 done

File: vendor/bat-native-ads/README.md (hunk 3 above)
Comment (tmancey) [reply vendor/bat-native-ads/README.md]: 👍 done. Did you not know this was coming in c++20

**Pull request merged**